### PR TITLE
Send user address to northstar

### PR DESCRIPTION
**Pull request by itsjoekent**
#### What's this PR do?
This PR updates the signup data form to push the user address data to Northstar. I re-used some of my logic from #7412 and also updated the helper text of the shipment module so nobody ever uses that again... I also fixed some spacing in the places I was touching the code because I was losing my mind.

Here is a profile I updated via the form

screen shot 2017-06-14 at 3 50 55 pm

### CODE REVIEW

File: lib/modules/dosomething/dosomething_user/dosomething_user.address.inc [hunk 1]
```diff
@@ -145,9 +149,35 @@ function dosomething_user_save_address_values($values) {
...
+
+  foreach ($address_fields as $drupal_key => $northstar_key) {
+    $field = $submission[$drupal_key];
+
+    if ($field) {
+      $profile[$northstar_key] = $field;
+    }
+  }
+
+  dosomething_northstar_client()->asClient()->put('v1/users/id/' . $northstar_id, $profile);
```
Comment (DFurnes) [inline lib/modules/dosomething/dosomething_user/dosomething_user.address.inc]: Let's do this with the logged-in user's access token and the `/profile` endpoint.

```php
dosomething_northstar_client()->post('/profile', $profile);
```

Comment (itsjoekent) [reply lib/modules/dosomething/dosomething_user/dosomething_user.address.inc]: @DFurnes Hm, this doesn't appear to work. I added the missing `v1/` from the example but I get the following error,

```
DoSomething\Gateway\Exceptions\UnauthorizedException: Exception in Northstar "POST v1/profile" endpoint: [401] {"error":"access_denied","message":"The resource owner or authorization server denied the request.","hint":"Requires the `user` scope."} in DoSomething\Gateway\Northstar->handleUnauthorizedException() (line 379 of /var/www/dev.dosomething.org/vendor/dosomething/gateway/src/AuthorizesWithNorthstar.php).
```

Comment (itsjoekent) [reply lib/modules/dosomething/dosomething_user/dosomething_user.address.inc]: I also just gave this a try but doesn't seem to make a difference 

```
dosomething_northstar_client()->asUser()->post('v1/profile', $profile);
```

Comment (DFurnes) [reply lib/modules/dosomething/dosomething_user/dosomething_user.address.inc]: Oh, we probably don't need to be requiring the `user` scope there. Let's fix that in Northstar.

**Comment by itsjoekent** [comment]
switched back to profile endpoint, works now

screen shot 2017-06-19 at 10 03 47 am

**Review by DFurnes — APPROVED** [review]

**Comment by itsjoekent** [comment]
fixes #7287

**Pull request merged**